feature/platform/transceiver/tests/zr_low_power_mode_test/zr_low_power_mode_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zr_low_power_mode_test
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/samplestream"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+)
+
+const (
+	samplingInterval     = 10 * time.Second
+	targetOutputPowerdBm = -10
+	targetFrequencyHz    = 193100000
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// validateStreamOutput validates that the OC path is streamed in the most recent subscription interval.
+func validateStreamOutput(t *testing.T, dut *ondatra.DUTDevice, streams map[string]*samplestream.SampleStream[string]) {
+	for key, stream := range streams {
+		output := stream.Next(t)
+		if output == nil {
+			t.Fatalf("OC path for %s not streamed in the most recent subscription interval", key)
+		}
+		value, ok := output.Val()
+		if !ok {
+			t.Fatalf("Error capturing streaming value for %s", key)
+		}
```

Comment:
Do we also need the check on the value `Kind` here? ("No" is a fair answer here, just curious.)

Reply:
Initially thought we don't need it since we are checking for empty string. But I suppose we could have better error messaging if we ad the check on value Kind here.